mempool/src/pool/store.rs

```diff
@@ -77,10 +85,21 @@ pub struct MempoolStore {
     pub(super) spender_txs: BTreeMap<OutPoint, Id<Transaction>>,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub(super) enum MempoolRemovalReason {
```

Comment:
I'm confused... why does this even exist?

Reply:
If you wanna keep this, please provide full documentation on it why it exists and what every arm does

Reply:
I copied this enum from Bitcoin Core. They have a mechanism for notifying clients interested in knowing when a transaction was removed for a reason other than inclusion in a block. Also, the Bitcoin Core wallet has some logic for transactions removed from the mempool due to conflicting transactions in a new block.
If this enum is confusing then I think there's no need for it at the moment. Replaced it with a boolean `remove_for_block`, which is all we really need in our logic.

Reply:
Booleans are usually a downgrade. Just some documentation helps.

Reply:
Ok, restored it and added some docs.